src/analysis/inner_product_space/adjoint.lean

```diff
@@ -0,0 +1,122 @@
+/-
+Copyright (c) 2021 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis, Heather Macbeth
+-/
+
+import analysis.inner_product_space.dual
+
+/-!
+# Adjoint of operators on Hilbert spaces
+
+Given an operator `A : E →L[𝕜] F`, where `E` and `F` are Hilbert spaces, its adjoint
+`adjoint A : F →L[𝕜] E` is the unique operator such that `⟪x, A y⟫ = ⟪adjoint A x, y⟫` for all
+`x` and `y`.
+
+We then use this to put a star algebra structure on `E →L[𝕜] E` with the adjoint as the star
+operation.
+
+## Implementation notes
+
+* The adjoint is defined as a conjugate-linear isometric equivalence between `E →L[𝕜] F` and
+  `F →L[𝕜] E`. The bare function `adjoint'` is only an intermediate definition and is not meant
+  to be used outside this file.
+
+## TODO
+
+* Prove the C⋆ property for `E →L[𝕜] E` to show that it is a C⋆-algebra.
+
+## Tags
+
+adjoint
+
+-/
+
+noncomputable theory
+open inner_product_space continuous_linear_map
+open_locale complex_conjugate
+
+variables {𝕜 E F G : Type*} [is_R_or_C 𝕜]
+variables [inner_product_space 𝕜 E] [inner_product_space 𝕜 F] [inner_product_space 𝕜 G]
+variables [complete_space E] [complete_space G]
+
+local notation `⟪`x`, `y`⟫` := @inner 𝕜 _ _ x y
+
+namespace continuous_linear_map
+
+/-- The adjoint, as a bare function. This is only meant as an auxiliary definition for
+the main definition `adjoint`, where this is bundled as a conjugate-linear isometric
+equivalence. -/
+def adjoint' : (E →L[𝕜] F) →L⋆[𝕜] F →L[𝕜] E :=
+(continuous_linear_map.compSL _ _ _ _ _ ((to_dual 𝕜 E).symm : normed_space.dual 𝕜 E →L⋆[𝕜] E)).comp
+  (to_sesq_form : (E →L[𝕜] F) →L[𝕜] F →L⋆[𝕜] normed_space.dual 𝕜 E)
+
+@[simp] lemma adjoint'_apply {A : E →L[𝕜] F} {x : F} :
+  adjoint' A x = ((to_dual 𝕜 E).symm : (normed_space.dual 𝕜 E) → E) ((to_sesq_form A) x) := rfl
+
+lemma adjoint'_inner_left {A : E →L[𝕜] F} {x : E} {y : F} : ⟪adjoint' A y, x⟫ = ⟪y, A x⟫ :=
+by { simp only [adjoint'_apply, to_dual_symm_apply, to_sesq_form_apply_coe, coe_comp',
+                innerSL_apply_coe]}
+
+lemma adjoint'_inner_right {A : E →L[𝕜] F} {x : E} {y : F} : ⟪x, adjoint' A y⟫ = ⟪A x, y⟫ :=
+by rw [←inner_conj_sym, adjoint'_inner_left, inner_conj_sym]
+
+variables [complete_space F]
+
+lemma adjoint'_adjoint'_apply (A : E →L[𝕜] F) : adjoint' (adjoint' A) = A :=
```

Comment:
Is there motivation for making `A` explicit here and implicit elsewhere? My intuition would be to use explicit in most places (e.g. for `A`, `x`, and `y` above), but I get there are reasons to do otherwise.

Reply:
```suggestion
lemma adjoint'_adjoint' (A : E →L[𝕜] F) : adjoint' (adjoint' A) = A :=
```
? `adjoint'_apply` above applies to an additional argument.

Reply:
In fact, for other lemmas like `adjoint'_apply`, `adjoint(')_inner_right` I would also be inclined to make all arguments explicit.  What do you think?

Reply:
Sounds good, I'll make the change.